test/test/lock/LockTests.java

```diff
@@ -10,6 +10,34 @@
 import one.profiler.test.TestProcess;
 
 public class LockTests {
+    private static void contendedLocks(TestProcess p, int threshold, double minRatio) throws Exception {
+        Output out = p.profile("-e lock --lock " + threshold + " --threads -o collapsed");
+        Output waitingLocks = out.filter("LockProfiling\\.contend;java\\.lang\\.Object");
+
+        double ratio = 100 * out.ratio("contend-0.5-", "contend-0.05-");
+
+        System.out.println("threshold = " + threshold + ", ratio = " + ratio + ", minRatio = " + minRatio);
+
+        assert (ratio >= minRatio) || (Double.isNaN(ratio) && Double.isNaN(minRatio));
+    }
+
+    // 0 is equivalent to disabling sampling of locks, so all profiles are included.
+    @Test(mainClass = LockProfiling.class, inputs = { "0", "75" })
+    @Test(mainClass = LockProfiling.class, inputs = { "1", "75" })
```

Comment:
Technically, there is no difference between `0` and `1`, you can leave only `0` case here.